lib/src/widgets/animated_gap.dart

```diff
@@ -0,0 +1,118 @@
+///
+///  * [SliverAnimatedGap], the [RenderSliver] version of this widget.
+class AnimatedGap extends ImplicitlyAnimatedWidget {
+  /// Creates a [Gap] that animates its parameters implicitly.
+  ///
+  /// The [curve] and [duration] arguments must not be null.
+  const AnimatedGap(
+    this.mainAxisExtent, {
+    Key? key,
+    this.crossAxisExtent,
+    this.color,
+    Curve curve = Curves.linear,
+    required Duration duration,
+    VoidCallback? onEnd,
+  })  : assert(mainAxisExtent >= 0 && mainAxisExtent < double.infinity),
+        assert(crossAxisExtent == null || crossAxisExtent >= 0),
+        super(
+          key: key,
+          curve: curve,
+          duration: duration,
+          onEnd: onEnd,
+        );
+
+  /// The amount of space this widget takes in the direction of its parent.
+  ///
+  /// For example:
+  /// - If the parent is a [Column] this is the height of this widget.
+  /// - If the parent is a [Row] this is the width of this widget.
+  ///
+  /// Must not be null and must be positive.
+  final double mainAxisExtent;
+
+  /// The amount of space this widget takes in the opposite direction of the
+  /// parent.
+  ///
+  /// For example:
+  /// - If the parent is a [Column] this is the width of this widget.
+  /// - If the parent is a [Row] this is the height of this widget.
+  ///
+  /// Must be positive or null. If it's null (the default) the cross axis extent
+  /// will be the same as the constraints of the parent in the opposite
+  /// direction.
+  final double? crossAxisExtent;
+
+  /// The color used to fill the gap.
+  final Color? color;
+
+  @override
+  AnimatedWidgetBaseState<AnimatedGap> createState() => _AnimatedGapState();
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(DoubleProperty('mainAxisExtent', mainAxisExtent));
+    properties.add(
+      DoubleProperty('crossAxisExtent', crossAxisExtent, defaultValue: null),
+    );
+    properties.add(ColorProperty('color', color, defaultValue: null));
+  }
+}
+
+class _AnimatedGapState extends AnimatedWidgetBaseState<AnimatedGap> {
+  Tween<double>? _mainAxisExtent;
+  Tween<double>? _crossAxisExtent;
+  ColorTween? _color;
+
+  @override
+  void forEachTween(TweenVisitor<dynamic> visitor) {
+    _mainAxisExtent = visitor(
+      _mainAxisExtent,
+      widget.mainAxisExtent,
+      (dynamic value) => Tween<double>(begin: value as double),
+    ) as Tween<double>?;
+
+    _crossAxisExtent = visitor(
+      _crossAxisExtent,
+      widget.crossAxisExtent,
+      (dynamic value) => Tween<double>(begin: value as double),
+    ) as Tween<double>?;
+
+    _color = visitor(
+      _color,
+      widget.color,
+      (dynamic value) => ColorTween(begin: value as Color),
+    ) as ColorTween?;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final mainAxisExtent =
+        _mainAxisExtent!.evaluate(animation).clamp(0.0, double.infinity);
```

Comment:
Let's remove this `!` I don't want to see 🙈

Reply:
It's based on what Flutter is doing for it's implicitly animated widgets, what I can do is instead returning 0 as a default value.